internal/chain/create_token_account.go

```diff
@@ -14,60 +14,43 @@ type CreateTokenAccount struct{}
 
 func (CreateTokenAccount) Type() types.TxType { return types.TxCreateTokenAccount }
 
-func checkCreateTokenAccount(st *StateManager, tx *transactions.GenTransaction) (accountUrl, tokenUrl, keyBookUrl *url.URL, err error) {
+func (CreateTokenAccount) Validate(st *StateManager, tx *transactions.GenTransaction) error {
 	body := new(protocol.TokenAccountCreate)
-	err = tx.As(body)
+	err := tx.As(body)
 	if err != nil {
-		return nil, nil, nil, fmt.Errorf("invalid payload: %v", err)
+		return fmt.Errorf("invalid payload: %v", err)
 	}
 
-	accountUrl, err = url.Parse(body.Url)
+	accountUrl, err := url.Parse(body.Url)
 	if err != nil {
-		return nil, nil, nil, fmt.Errorf("invalid account URL: %v", err)
+		return fmt.Errorf("invalid account URL: %v", err)
 	}
 
-	tokenUrl, err = url.Parse(body.TokenUrl)
+	tokenUrl, err := url.Parse(body.TokenUrl)
 	if err != nil {
-		return nil, nil, nil, fmt.Errorf("invalid token URL: %v", err)
+		return fmt.Errorf("invalid token URL: %v", err)
 	}
 	// TODO Make sure tokenUrl is a real kind of token
 
 	if !accountUrl.Identity().Equal(st.SponsorUrl) {
-		return nil, nil, nil, fmt.Errorf("%q cannot sponsor %q", st.SponsorUrl, accountUrl)
+		return fmt.Errorf("%q cannot sponsor %q", st.SponsorUrl, accountUrl)
 	}
 
-	if body.KeyBookUrl != "" {
-		keyBookUrl, err = url.Parse(body.KeyBookUrl)
+	account := state.NewTokenAccount(accountUrl.String(), tokenUrl.String())
+	if body.KeyBookUrl == "" {
+		account.SigSpecId = st.Sponsor.Header().SigSpecId
```

Comment:
Maybe not a fix for this PR, but while I'm thinking about it, sigspecid should be keybookid or keybookurl

Reply:
I intend to do this as part of [AC-533](https://accumulate.atlassian.net/browse/AC-533)